### DISCUSSION TITLE
visit__default.default is not a function

### DISCUSSION THREAD

**Discussion by ThomasKellyOV** [facebook/docusaurus#9652] [Q&A] [answered]
Hi, I am attempting to migrate my docusaurus project to v3, however I am receiving an unexpected error when running the `docusaurus start` command. This command throws the following error for several .md and .mdx files in my project:

```
Error: MDX compilation failed for file "file/path/*.md"
Cause: visit__default.default is not a function
Details:
TypeError: visit__default.default is not a function
```
I have installed and run the MDX checker suggested in [this blog post](https://docusaurus.io/blog/preparing-your-site-for-docusaurus-v3#using-the-mdx-checker-cli), however the checker does not list any compilation errors for the given file. Most of the files that the error is thrown for do not contain any jsx, so I dont know what could be causing the issue. Does anyone happen to recognize this error or have any troubleshooting suggestions?

**Comment by slorber** [comment]
please create a smaller repro, it's impossible to help you without seeing a docusaurus config, dependencies, and the content of the markdown files

Most likely a remark plugin you use is not compatible with v3, is not set up properly or does not use the correct version.

If you use packages such as `unist-util-visit` and other similar things, make sure they are upgraded and read their changelog to ensure their are compatible with MDX v2

**Comment by ThomasKellyOV** [comment, marked as answer]
Just in case anyone else runs into this error, it was caused by a mdx-mermaid dependency and was resolved by upgrading to v2.0.0

**Reply by saitaiky** [reply]
Thank you!
